MediaGalleryMetadataApi/Model/ExtractMetadataComposite.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryMetadataApi\Model;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\MediaGalleryMetadataApi\Api\Data\MetadataInterface;
+use Magento\MediaGalleryMetadataApi\Api\ExtractMetadataInterface;
+use Magento\MediaGalleryMetadataApi\Api\Data\MetadataInterfaceFactory;
+
+/**
+ * Metadata extractor composite
+ */
+class ExtractMetadataComposite implements ExtractMetadataInterface
+{
+    /**
+     * @var ExtractMetadataInterface[]
+     */
+    private $extractors;
+
+    /**
+     * @var MetadataInterfaceFactory
+     */
+    private $metadataFactory;
+
+    /**
+     * @param MetadataInterfaceFactory $metadataFactory
+     * @param ExtractMetadataInterface[] $extractors
+     */
+    public function __construct(
+        MetadataInterfaceFactory $metadataFactory,
+        array $extractors
+    ) {
+        $this->metadataFactory = $metadataFactory;
+        $this->extractors = $extractors;
```

Comment:
It would be good to add interface validation for extractors and addmetadata items as well, they should implement AddMetatdata/ExtractMetadataInterface

Reply:
ok